tests/models/test_implicit_bpr.py

```diff
@@ -241,11 +282,18 @@ def test_i2i(
             actual,
         )
 
-    @pytest.mark.skip("BPR doesn't behave deterministically")
     def test_second_fit_refits_model(self, dataset: Dataset, use_gpu: bool) -> None:
-        base_model = BayesianPersonalizedRanking(factors=8, num_threads=2, use_gpu=use_gpu, random_state=1)
+        # note that num_threads > 1 will make model training undeterministic
+        # https://github.com/benfred/implicit/issues/710
+        # GPU training is always undeterministic so we only test for CPU training
+        base_model = BayesianPersonalizedRanking(factors=8, num_threads=1, use_gpu=False, random_state=1)
```

Comment:
```suggestion
        # GPU training is always nondeterministic so we only test for CPU training
        if use_gpu:
            pytest.skip("BPR is nondeterministic on GPU")
        base_model = BayesianPersonalizedRanking(factors=8, num_threads=1, use_gpu=use_gpu, random_state=1)
```
(To avoid having 2 identical tests)